mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala

```diff
@@ -431,6 +515,13 @@ class SparseMatrix(
     }
   }
 
+  override def equals(o: Any) = o match {
```

Comment:
We need to override hashCode too. Let's move this out to another PR (with hashCode). We can merge that one first and then this UDT PR.